src/shared/components/Searchbar/Searchbar.tsx

```diff
@@ -1,87 +1,195 @@
+import { SearchBox } from './SearchBox'
+import { SearchHelper } from './Searchbar.style'
 import { CancelButton, Container, Input, SearchButton, StyledSvgOutlineSearch } from './Searchbar.style'
 
 type SearchbarProps = {
-  value: string
+  value: string | null
   onCancel?: () => void
   showCancelButton?: boolean
   controlled?: boolean
   onClick?: () => void
+  hasFocus: boolean
+  onClose: () => void
 } & React.DetailedHTMLProps<React.HTMLAttributes<HTMLInputElement>, HTMLInputElement>
-export const Searchbar: React.FC<SearchbarProps> = ({
-  placeholder,
-  onChange,
-  onFocus,
-  onCancel,
-  showCancelButton = false,
-  controlled = false,
-  value: externalValue,
-  onBlur,
-  onSubmit,
-  className,
-  onClick,
-  ...htmlProps
-}) => {
-  const [value, setValue] = useState('')
-  const mdMatch = useMediaMatch('md')
+export const Searchbar = React.forwardRef<HTMLDivElement, SearchbarProps>(
+  (
+    {
+      placeholder,
+      onChange,
+      onFocus,
+      onCancel,
+      value,
+      onBlur,
+      onSubmit,
+      className,
+      onClick,
+      hasFocus,
+      onClose,
+      onKeyDown,
+      ...htmlProps
+    },
+    ref
+  ) => {
+    const [recentSearch, setRecentSearch] = useState<string | null | undefined>(null)
+    const mdMatch = useMediaMatch('sm')
+    const inputRef = useRef<HTMLInputElement>(null)
+    const [selectedItem, setSelectedItem] = useState<number | null>(null)
+    const navigate = useNavigate()
+    const location = useLocation()
+    const locationState = location.state as RoutingState
+    const overlaidLocation = locationState?.overlaidLocation || location
+    const query = recentSearch || value
+    const { addRecentSearch } = usePersonalDataStore((state) => ({
+      addRecentSearch: state.actions.addRecentSearch,
+    }))
 
-  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-    if (onChange) {
-      onChange(e)
+    useEffect(() => {
+      const onKeyPress = (event: KeyboardEvent) => {
+        if (event.key === '/') {
+          onClick?.()
+          inputRef.current && setTimeout(() => inputRef.current?.focus(), 10)
+        }
+        if ((event.key === 'Enter' || event.key === 'NumpadEnter') && !!query) {
+          inputRef?.current?.blur()
+          onClose?.()
+        }
+      }
+      window.addEventListener('keydown', onKeyPress)
+      return () => {
+        window.removeEventListener('keydown', onKeyPress)
+      }
+    }, [onClick, onClose, query])
+
+    const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+      if ((event.key === 'Enter' || event.key === 'NumpadEnter') && query?.trim() && !selectedItem) {
+        const state: RoutingState = { overlaidLocation }
+        addRecentSearch(new Date().getTime(), query)
+
+        // navigate to search results
+        navigate(absoluteRoutes.viewer.search({ query: query?.trim() }), { state })
+      }
+      if (event.key === 'Escape' || event.key === 'Esc' || event.key === 'Tab') {
```

Comment:
```suggestion
      if (event.key === 'Escape' || event.key === 'Esc') {
```
I've noticed that when the focus is on search input you need to press TAB twice to switch to `start publishing`. Is it intended?  I think we could remove Tab from this line and pass `onBlur` to SearchBar instead. Wdyt?